src/schedule/dma_multi_chan_domain.c

```diff
@@ -222,39 +224,41 @@ static int dma_multi_chan_domain_unregister(struct ll_schedule_domain *domain,
 		return 0;
 
 	for (i = 0; i < dma_domain->num_dma; ++i) {
-		for (j = 0; j < dmas[i].plat_data.channels; ++j) {
-			/* channel not set as scheduling source */
-			if (!dma_is_scheduling_source(&dmas[i].chan[j]))
-				continue;
-
-			/* channel still running */
-			if (dmas[i].chan[j].status == COMP_STATE_ACTIVE)
-				continue;
-
-			/* channel owned by different core */
-			if (core != dmas[i].chan[j].core)
-				continue;
-
-			/* channel hasn't been running */
-			if (!(dma_domain->channel_mask[i][core] & BIT(j)))
-				continue;
-
-			dma_interrupt_legacy(&dmas[i].chan[j], DMA_IRQ_MASK);
-			dma_interrupt_legacy(&dmas[i].chan[j], DMA_IRQ_CLEAR);
-			interrupt_clear_mask(dma_domain->data[i][j].irq,
-					     BIT(j));
-
-			dma_domain->data[i][j].task = NULL;
-			dma_domain->channel_mask[i][core] &= ~BIT(j);
-
-			/* unregister interrupt */
-			if (!dma_domain->aggregated_irq)
-				dma_multi_chan_domain_irq_unregister(
-						&dma_domain->data[i][j]);
-			else if (!dma_domain->channel_mask[i][core])
-				dma_multi_chan_domain_irq_unregister(
-						dma_domain->arg[i][core]);
-			return 0;
+		if (dmas[i].chan != NULL) {
```

Comment:
Minor: you could also "continue" if chan is NULL and not have to modify and indent the whole loop.

Reply:
done